torch/csrc/jit/script/module.h

```diff
@@ -113,32 +103,31 @@ struct TORCH_API Method {
   // This is the _lowered_ function and is different than the
   // first-class function in class_compilation_unit()
   std::shared_ptr<Function> function_;
-
-  // parameters and attributes loaded from the Module and appending
-  // before calling function_
-  std::vector<Slot> initial_ivalues_;
-  FunctionSchema schema_;
 };
 
 struct Module;
 
 struct TORCH_API Module {
   TH_DISALLOW_COPY_AND_ASSIGN(Module);
-  Module()
-      : name_("__main__"),
+  Module(std::string class_name)
+      : field_name_("__main__"),
         module_value_(c10::ivalue::Object::create(
-            ClassType::createModuleType(std::make_shared<CompilationUnit>()),
+            ClassType::create(
+                QualifiedName(std::move(class_name)),
+                std::make_shared<CompilationUnit>(),
+                /*is_module=*/true),
             0)) {}
-
+  Module()
+  : Module("$Module") {}
   ~Module() {
     // ClassType own the compilation unit of their Functions, but each
     // Function has a self argument which owns the ClassType, created a
     // reference cycle. By dropping all the methods of the module's class
     // here we break the cycle.
     class_compilation_unit().drop_all_functions();
   }
-  const std::string& name() const {
-    return name_;
+  const std::string& field_name() const {
```

Comment:
field_name is a little confusing—unqualified_name?

Reply:
There are now two names, the name of the Class and the name of the module as a member of a parent module. We still rely on this "field_name" in a few places and it is never qualified.  There is a comment that explains this. My hope is that the "Module because a view patch" simply removes it.